code/modules/mob/living/carbon/xenomorph/attack_alien.dm

```diff
@@ -199,11 +199,13 @@
 				X.do_attack_animation(src, ATTACK_EFFECT_CLAW)
 				to_chat(X, span_warning("We disable the creatures hivemind sight apparatus."))
 				return FALSE
-
-		if(length(light_sources) || locate(/obj/effect/overlay/light_visible) in vis_contents)
+		var/lights_found = FALSE
+		for(var/atom/light AS in GLOB.nightfall_toggleable_lights)
+			if(light.loc == src && light.light_on && light.turn_light(null, FALSE, 0, TRUE))
+				lights_found = TRUE
```

Comment:
This performs terribly for what it does and doesnt even break you know better than to do this

Reply:
It shouldn't break?

And ok, but fix your shit maybe?